### PR TITLE
refactor(sbTag): :recycle: SbTag component SFC refactor

### PR DESCRIPTION
**Pull request by hazlr**
## Pull request type

Jira Link: [INT](url) [Not applicable]

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## How to test this PR

1. First build the Design System by running: `yarn storybook`
2. Check if the `<SbTag/>` component and all its variation are still working as previously
3. Verify all Unit Tests are still passing by running `yarn test:unit`

## What is the new behavior?

No new behavior has been added. The component instance creation has been refactored from a render function to a SFC (Single File Component).
A new unit test has been added to assert that when the component is instantiated without setting the `closable` prop to `true`. It should not have a `<SbIcon>` component child.

## Other information

N/A

### CODE REVIEW

**Comment by thiagosaife** [comment]
Hello @hazlr . Thanks for the awesome collaboration. We will fully check your pull request on next week.

File: src/components/Tag/Tag.stories.js [hunk 1]
```diff
@@ -61,9 +45,21 @@ export default {
...
-export const Default = BadgeTemplate.bind({})
+const getComponentOptions = (args) => ({
+  components: { SbTag },
+  props: Object.keys(args),
+  template: `
+    <SbTag
+      :closable="closable"
+      :label="label"
+      :type="type"
+      @close="onClick"
```
Comment (thiagosaife) [inline src/components/Tag/Tag.stories.js:56]: Just rename this method to `handleClose`, please.

**Review by gustavomelki — APPROVED** [review]
Looks good! Nice work!!!

**Review by thiagosaife — APPROVED** [review]
Nice job!

**Review by emanuelgsouza — APPROVED** [review]
Nice work

**Pull request merged**